twill-yarn/src/main/java/org/apache/twill/internal/appmaster/ApplicationMasterService.java

```diff
@@ -239,14 +359,8 @@ protected void doStop() throws Exception {
     LOG.info("Stop application master with spec: {}",
              TwillRuntimeSpecificationAdapter.create().toJson(twillRuntimeSpec));
 
-    if (eventHandler != null) {
-      try {
-        // call event handler destroy. If there is error, only log and not affected stop sequence.
-        eventHandler.destroy();
-      } catch (Throwable t) {
-        LOG.warn("Exception when calling {}.destroy()", eventHandler.getClass().getName(), t);
-      }
-    }
+    // call event handler destroy
+    eventHandler.destroy();
```

Comment:
Shouldn't call `destroy()` here anymore. It should be called after call other calls to the event handler, as it is acted as a cleanup call.